src/types/state.js

```diff
@@ -124,6 +119,22 @@ export type AppState = {|
   +lastVisibleThreadTabSlug: TabSlug,
 |};
 
+export type UploadPhase = 'local' | 'uploading' | 'uploaded' | 'error';
```

Comment:
I'd like to move to something like:

```
export type UploadState = {|
  phase: 'local',
  url: string,
|} | {| ... |};
```

I filed a follow-up: https://github.com/firefox-devtools/profiler/issues/1894